sled-agent/config-reconciler/src/dataset_serialization_task.rs

```diff
@@ -83,6 +87,29 @@ pub enum DatasetEnsureError {
     TestError(&'static str),
 }
 
+impl DatasetEnsureError {
+    fn is_retryable(&self) -> bool {
+        match self {
+            // Errors that we don't know for sure _aren't_ retryable.
+            DatasetEnsureError::ZpoolNotFound(_)
+            | DatasetEnsureError::EnsureFailed { .. } => true,
```

Comment:
This comment doesn't seem to align with the return value?

Comment says "for sure are not retryable" but returning "is_retryable() -> true"

Reply:
I think (?) the comment is correct, but has too many negatives to be understandable. (If we did know for sure that these aren't retryable, we ought to return `false`. But we don't know that, so we have to return `true`.)

I'll rephrase to something like: 

> These errors _might_ be retryable; there are probably cases where they won't be, but we need more context than we have available from just the error to know for sure. For now, assume they are retryable - that may mean we churn on something doomed, but that's better than failing to retry something we should have retried.

Reply:
ah, yeah, I misread that from the double negative. The clarified comment would be nice